// angular-bootstrap-md/esm5/lib/free/breadcrumbs/mdb-breadcrumb.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
var MdbBreadcrumbComponent = /** @class */ (function () {
    function MdbBreadcrumbComponent() {
    }
    MdbBreadcrumbComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mdb-breadcrumb',
                    template: "<ol class=\"breadcrumb list-inline list-unstyled {{customClass}} text-{{textTransform}}\">\n  <ng-content></ng-content>\n</ol>\n",
                    changeDetection: ChangeDetectionStrategy.OnPush
                }] }
    ];
    MdbBreadcrumbComponent.propDecorators = {
        customClass: [{ type: Input }],
        textTransform: [{ type: Input }]
    };
    return MdbBreadcrumbComponent;
}());
export { MdbBreadcrumbComponent };
if (false) {
    /** @type {?} */
    MdbBreadcrumbComponent.prototype.customClass;
    /** @type {?} */
    MdbBreadcrumbComponent.prototype.textTransform;
}
