// angular-bootstrap-md/esm5/lib/free/utils/keyboard-navigation.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @license
 * Copyright Google LLC All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */
/** @type {?} */
export var MAC_ENTER = 3;
/** @type {?} */
export var BACKSPACE = 8;
/** @type {?} */
export var TAB = 9;
/** @type {?} */
export var NUM_CENTER = 12;
/** @type {?} */
export var ENTER = 13;
/** @type {?} */
export var SHIFT = 16;
/** @type {?} */
export var CONTROL = 17;
/** @type {?} */
export var ALT = 18;
/** @type {?} */
export var PAUSE = 19;
/** @type {?} */
export var CAPS_LOCK = 20;
/** @type {?} */
export var ESCAPE = 27;
/** @type {?} */
export var SPACE = 32;
/** @type {?} */
export var PAGE_UP = 33;
/** @type {?} */
export var PAGE_DOWN = 34;
/** @type {?} */
export var END = 35;
/** @type {?} */
export var HOME = 36;
/** @type {?} */
export var LEFT_ARROW = 37;
/** @type {?} */
export var UP_ARROW = 38;
/** @type {?} */
export var RIGHT_ARROW = 39;
/** @type {?} */
export var DOWN_ARROW = 40;
/** @type {?} */
export var PLUS_SIGN = 43;
/** @type {?} */
export var PRINT_SCREEN = 44;
/** @type {?} */
export var INSERT = 45;
/** @type {?} */
export var DELETE = 46;
/** @type {?} */
export var ZERO = 48;
/** @type {?} */
export var ONE = 49;
/** @type {?} */
export var TWO = 50;
/** @type {?} */
export var THREE = 51;
/** @type {?} */
export var FOUR = 52;
/** @type {?} */
export var FIVE = 53;
/** @type {?} */
export var SIX = 54;
/** @type {?} */
export var SEVEN = 55;
/** @type {?} */
export var EIGHT = 56;
/** @type {?} */
export var NINE = 57;
/** @type {?} */
export var FF_SEMICOLON = 59;
// Firefox (Gecko) fires this for semicolon instead of 186
/** @type {?} */
export var FF_EQUALS = 61;
// Firefox (Gecko) fires this for equals instead of 187
/** @type {?} */
export var QUESTION_MARK = 63;
/** @type {?} */
export var AT_SIGN = 64;
/** @type {?} */
export var A = 65;
/** @type {?} */
export var B = 66;
/** @type {?} */
export var C = 67;
/** @type {?} */
export var D = 68;
/** @type {?} */
export var E = 69;
/** @type {?} */
export var F = 70;
/** @type {?} */
export var G = 71;
/** @type {?} */
export var H = 72;
/** @type {?} */
export var I = 73;
/** @type {?} */
export var J = 74;
/** @type {?} */
export var K = 75;
/** @type {?} */
export var L = 76;
/** @type {?} */
export var M = 77;
/** @type {?} */
export var N = 78;
/** @type {?} */
export var O = 79;
/** @type {?} */
export var P = 80;
/** @type {?} */
export var Q = 81;
/** @type {?} */
export var R = 82;
/** @type {?} */
export var S = 83;
/** @type {?} */
export var T = 84;
/** @type {?} */
export var U = 85;
/** @type {?} */
export var V = 86;
/** @type {?} */
export var W = 87;
/** @type {?} */
export var X = 88;
/** @type {?} */
export var Y = 89;
/** @type {?} */
export var Z = 90;
/** @type {?} */
export var META = 91;
// WIN_KEY_LEFT
/** @type {?} */
export var MAC_WK_CMD_LEFT = 91;
/** @type {?} */
export var MAC_WK_CMD_RIGHT = 93;
/** @type {?} */
export var CONTEXT_MENU = 93;
/** @type {?} */
export var NUMPAD_ZERO = 96;
/** @type {?} */
export var NUMPAD_ONE = 97;
/** @type {?} */
export var NUMPAD_TWO = 98;
/** @type {?} */
export var NUMPAD_THREE = 99;
/** @type {?} */
export var NUMPAD_FOUR = 100;
/** @type {?} */
export var NUMPAD_FIVE = 101;
/** @type {?} */
export var NUMPAD_SIX = 102;
/** @type {?} */
export var NUMPAD_SEVEN = 103;
/** @type {?} */
export var NUMPAD_EIGHT = 104;
/** @type {?} */
export var NUMPAD_NINE = 105;
/** @type {?} */
export var NUMPAD_MULTIPLY = 106;
/** @type {?} */
export var NUMPAD_PLUS = 107;
/** @type {?} */
export var NUMPAD_MINUS = 109;
/** @type {?} */
export var NUMPAD_PERIOD = 110;
/** @type {?} */
export var NUMPAD_DIVIDE = 111;
/** @type {?} */
export var F1 = 112;
/** @type {?} */
export var F2 = 113;
/** @type {?} */
export var F3 = 114;
/** @type {?} */
export var F4 = 115;
/** @type {?} */
export var F5 = 116;
/** @type {?} */
export var F6 = 117;
/** @type {?} */
export var F7 = 118;
/** @type {?} */
export var F8 = 119;
/** @type {?} */
export var F9 = 120;
/** @type {?} */
export var F10 = 121;
/** @type {?} */
export var F11 = 122;
/** @type {?} */
export var F12 = 123;
/** @type {?} */
export var NUM_LOCK = 144;
/** @type {?} */
export var SCROLL_LOCK = 145;
/** @type {?} */
export var FIRST_MEDIA = 166;
/** @type {?} */
export var FF_MINUS = 173;
/** @type {?} */
export var MUTE = 173;
// Firefox (Gecko) fires 181 for MUTE
/** @type {?} */
export var VOLUME_DOWN = 174;
// Firefox (Gecko) fires 182 for VOLUME_DOWN
/** @type {?} */
export var VOLUME_UP = 175;
// Firefox (Gecko) fires 183 for VOLUME_UP
/** @type {?} */
export var FF_MUTE = 181;
/** @type {?} */
export var FF_VOLUME_DOWN = 182;
/** @type {?} */
export var LAST_MEDIA = 183;
/** @type {?} */
export var FF_VOLUME_UP = 183;
/** @type {?} */
export var SEMICOLON = 186;
// Firefox (Gecko) fires 59 for SEMICOLON
/** @type {?} */
export var EQUALS = 187;
// Firefox (Gecko) fires 61 for EQUALS
/** @type {?} */
export var COMMA = 188;
/** @type {?} */
export var DASH = 189;
// Firefox (Gecko) fires 173 for DASH/MINUS
/** @type {?} */
export var SLASH = 191;
/** @type {?} */
export var APOSTROPHE = 192;
/** @type {?} */
export var TILDE = 192;
/** @type {?} */
export var OPEN_SQUARE_BRACKET = 219;
/** @type {?} */
export var BACKSLASH = 220;
/** @type {?} */
export var CLOSE_SQUARE_BRACKET = 221;
/** @type {?} */
export var SINGLE_QUOTE = 222;
/** @type {?} */
export var MAC_META = 224;
